File: Lodjinha/app/src/main/java/freitas/dias/lodjinha/ui/activity/BaseActivity.java
package freitas.dias.lodjinha.ui.activity;


import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {

    protected void replaceFragment(@IdRes int containerViewId, @NonNull Fragment fragment,
                                   @NonNull String fragmentTag, @Nullable String backStackStateName) {
        getSupportFragmentManager().beginTransaction().setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
                .replace(containerViewId, fragment, fragmentTag).addToBackStack(backStackStateName)
                .commit();
    }
}

File: app/src/main/java/com/caio/lodjinha/base/BaseActivity.kt
package com.caio.lodjinha.base

import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import com.caio.lodjinha.R
import com.caio.lodjinha.di.ApplicationBase

open class BaseActivity : AppCompatActivity() {

    init {
        ApplicationBase.activityContext = this
    }

    fun addFragment(fragment: Fragment) {

        supportFragmentManager
            .beginTransaction()
            .replace(R.id.content, fragment, fragment.javaClass.simpleName)
            .setCustomAnimations(R.anim.abc_fade_in, R.anim.abc_fade_out)
            .addToBackStack(null)
            .commit()
    }
}